src/main/java/it/gov/pagopa/reward/service/lookup/HpanInitiativesServiceImpl.java

```diff
@@ -104,17 +84,17 @@ private HpanInitiatives deletedHpanInitiative(HpanInitiatives hpanInitiatives, H
                 log.trace("[DELETED_HPAN] [HPAN_WITH_INITIATIVE] [INITIATIVE_IS_PRESENT] The hpan: %s, contain the initiative: %s".formatted(dto.getHpan(),dto.getInitiativeId()));
```

Comment:
I'm writing here the review for the delete operation

at row 81: store the "find" execution and reuse the object instead of recalculate it

at row 87: use the "max" stream operator to obtain the object having the max activeStart

at row 92: the lastActiveInterval.getEndInterval() != null is redundant

at row 96: we should not create a new interval, but modified the latest in order to fill it's end date!

at row 105: this should be an unexpected case, let's log using ERROR level